xprof/tally.cpp

```diff
@@ -32,9 +34,60 @@ void print_metadata(std::vector<std::string> metadata) {
     std::cout << value << std::endl;
 }
 
+void get_backend_id_from_name(const char *name, int *identifier){
+  *identifier = -1;
+  for(int i = 0; i < BACKEND_MAX; ++i)
+    // backend_name is located in xprof_utils.cpp
+    if (strcmp(backend_name[i],name) == 0) *identifier = i;
+}
+
+// Reference: https://favtutor.com/blogs/split-string-cpp
+std::vector <std::string> split(std::string str, char separator) {
+  std::vector <std::string> strings;
+  int startIndex = 0, endIndex = 0;
+  for (unsigned i = 0; i <= str.size(); i++) {
+    // If we reached the end of the word or the end of the input.
+    if (str[i] == separator || i == str.size()) {
+        endIndex = i;
+        std::string temp;
+        temp.append(str, startIndex, endIndex - startIndex);
+        strings.push_back(temp);
+        startIndex = endIndex + 1;
+    }
+  }
+
+  return strings;
+}
+
 void btx_initialize_usr_data(void *btx_handle, void **usr_data) {
   /* User allocates its own data structure */
   *usr_data = new tally_dispatch_t;
+  tally_dispatch_t *data = (tally_dispatch_t *)(*usr_data);
```

Comment:
I think is clearer: 
```suggestion
  btx_read_params *data = new tally_dispatch_t;
  *usr_data = data;
```
(less cast)